infra/main.tf

```diff
@@ -435,293 +435,102 @@ resource "kubernetes_service" "ukbb_rb_static" {
-}
-
-resource "google_service_account" "query" {
-  account_id = "query-${random_id.query_name_suffix.hex}"
-}
-
-resource "google_service_account_key" "query_key" {
-  service_account_id = google_service_account.query.name
-}
-
-resource "kubernetes_secret" "query_gsa_key" {
-  metadata {
-    name = "query-gsa-key"
-  }
-
-  data = {
-    "key.json" = base64decode(google_service_account_key.query_key.private_key)
-  }
+module "query_gsa_secret" {
+  source = "./gsa_k8s_secret"
+  name = "query"
 }
 
 resource "google_storage_bucket_iam_member" "query_hail_query_bucket_storage_admin" {
   bucket = google_storage_bucket.hail_query.name
   role = "roles/storage.admin"
-  member = "serviceAccount:${google_service_account.query.email}"
+  member = "serviceAccount:${module.query_gsa_secret.email}"
 }
 
 resource "google_storage_bucket_iam_member" "batch_hail_query_bucket_storage_viewer" {
   bucket = google_storage_bucket.hail_query.name
   role = "roles/storage.objectViewer"
-  member = "serviceAccount:${google_service_account.batch.email}"
+  member = "serviceAccount:${module.batch_gsa_secret.email}"
 }
 
-resource "google_service_account" "benchmark" {
-  account_id = "benchmark"
+module "benchmark_gsa_secret" {
+  source = "./gsa_k8s_secret"
+  name = "benchmark"
 }
 
-resource "google_service_account_key" "benchmark_key" {
-  service_account_id = google_service_account.benchmark.name
-}
-
-resource "kubernetes_secret" "benchmark_gsa_key" {
-  metadata {
-    name = "benchmark-gsa-key"
-  }
-
-  data = {
-    "key.json" = base64decode(google_service_account_key.benchmark_key.private_key)
-  }
-}
-
-resource "random_id" "ci_name_suffix" {
-  byte_length = 2
-}
-
-resource "google_service_account" "ci" {
-  account_id = "ci-${random_id.ci_name_suffix.hex}"
+module "ci_gsa_secret" {
+  source = "./gsa_k8s_secret"
+  name = "ci"
 }
 
 resource "google_artifact_registry_repository_iam_member" "artifact_registry_viewer" {
   provider = google-beta
   repository = google_artifact_registry_repository.repository.name
   location = var.gcp_location
   role = "roles/artifactregistry.reader"
-  member = "serviceAccount:${google_service_account.ci.email}"
-}
-
-resource "google_service_account_key" "ci_key" {
-  service_account_id = google_service_account.ci.name
-}
-
-resource "kubernetes_secret" "ci_gsa_key" {
-  metadata {
-    name = "ci-gsa-key"
-  }
-
-  data = {
-    "key.json" = base64decode(google_service_account_key.ci_key.private_key)
-  }
+  member = "serviceAccount:${module.ci_gsa_secret.email}"
 }
 
-resource "google_service_account" "monitoring" {
-  account_id = "monitoring"
+module "monitoring_gsa_secret" {
+  source = "./gsa_k8s_secret"
+  name = "monitoring"
 }
 
-resource "google_service_account_key" "monitoring_key" {
-  service_account_id = google_service_account.monitoring.name
+module "test_gsa_secret" {
+  source = "./gsa_k8s_secret"
+  name = "test"
+  iam_roles = [
+    "compute.instanceAdmin.v1",
+    "iam.serviceAccountUser",
```

Comment:
Can you remind me why the test user needs access to all of these roles? Is this the service account that's being used for the non-default batch driver instead of batch2-agent?

Reply:
I actually don't know the answer to this, I was just translating from Cotton's terraform. It could be that we can remove some of these.